src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -33,6 +33,12 @@ enum class JoinHashBuildMode { AllPositions, ExistenceOnly };
 
 using Hash = size_t;
 
+template <typename T>
+PolymorphicAllocator<T> alloc(const std::string& operator_data_structure) {
+  return PolymorphicAllocator<T>{
+      Hyrise::get().memory_resource_manager.get_memory_resource(OperatorType::JoinHash, operator_data_structure)};
```

Comment:
Can this be a method of `AbstractOperator`? `AbstractOperator::_alloc(...)` 

```c++
template <typename T>
PolymorphicAllocator<T> _alloc(const std::string& operator_data_structure) {
  return PolymorphicAllocator<T>{
      Hyrise::get().memory_resource_manager.get_memory_resource(_type, operator_data_structure)};
}
```

Reply:
This is an amazing idea, thanks for the suggestion!